dataprofiler/labelers/column_name_model.py

```diff
@@ -0,0 +1,258 @@
+                    """`{}` must be a list of dictionaries with at
+                    least 'attribute' as the key""".format(
+                        param
+                    )
+                )
+            elif param == "true_positive_dict" and (
+                not isinstance(value, list)
+                or not isinstance(value[0], dict)
+                and "attribute" not in value[0].keys()
+                and "list" not in value[0].keys()
+            ):
+                errors.append(
+                    """`{}` must be a list of dictionaries each with the following
+                    two keys: 'attribute' and 'label'""".format(
+                        param
+                    )
+                )
+            elif param not in list_of_accepted_parameters:
+                errors.append("`{}` is not an accepted parameter.".format(param))
+        if errors:
+            raise ValueError("\n".join(errors))
+
+    @staticmethod
+    def _make_lower_case(str, **kwargs):
+        return str.lower()
+
+    def _compare_negative(
+        self, list_of_column_names, check_values_dict, negative_threshold
+    ):
+        """Filter out column name examples that are false positives."""
+        scores = self._model(
+            list_of_column_names,
+            check_values_dict,
+            self._make_lower_case,
+            rapidfuzz.fuzz.token_sort_ratio,
+        )
+
+        list_of_column_names_filtered = []
+        for i in range(len(list_of_column_names)):
+            if scores[i][0] < negative_threshold:
+                list_of_column_names_filtered.append(list_of_column_names[i])
+
+        return list_of_column_names_filtered
+
+    def _compare_positive(
+        self,
+        list_of_column_names,
+        check_values_dict,
+        positive_threshold,
+        include_label,
+        show_confidences,
+    ):
+        """Calculate similarity scores for positive examples."""
+        scores = self._model(
+            list_of_column_names,
+            check_values_dict,
+            self._make_lower_case,
+            rapidfuzz.fuzz.token_sort_ratio,
+            include_label=include_label,
+        )
+
+        output_dictionary = {}
+        for i in range(len(list_of_column_names)):
+            if scores[i][0] > positive_threshold:
+                output_dictionary[list_of_column_names[i]] = {}
+                output_dictionary[list_of_column_names[i]]["pred"] = check_values_dict[
+                    scores[i][1]
+                ]["label"]
+                if show_confidences:
+                    output_dictionary[list_of_column_names[i]]["conf"] = scores[i][0]
+
+        return output_dictionary
+
+    def _construct_model(self):
+        pass
+
+    def _reconstruct_model(self):
+        pass
+
+    def _need_to_reconstruct_model(self):
+        pass
+
+    def reset_weights(self):
+        """Reset weights function."""
+        pass
+
+    @require_module(["rapidfuzz"])
```

Comment:
do we have a test for this?

Reply:
Also, I think this is going to spout a graph profiling error as oppose to a labeler error